containers/daap-delete-data-product/src/var/task/delete_data_product.py

```diff
@@ -24,26 +31,24 @@ def handler(event, context):
     """
 
     data_product_name = event["pathParameters"].get("data-product-name")
-
     logger.add_data_product(data_product_name)
     logger.info(f"event: {event}")
 
-    data_product_metadata = DataProductMetadata(
-        data_product_name=data_product_name, logger=logger, input_data=None
-    )
-    if not data_product_metadata.exists:
-        error_message = (
-            f"Could not locate metadata for data product: {data_product_name}."
-        )
-        logger.error(error_message)
-        return format_error_response(HTTPStatus.BAD_REQUEST, event, error_message)
-
-    version_creator = VersionCreator(data_product_name=data_product_name, logger=logger)
     try:
-        version_creator.remove_all_versions_of_data_product()
-    except ValueError as e:
-        return format_error_response(HTTPStatus.BAD_REQUEST, event, str(e))
-    else:
-        msg = f"Successfully removed data product '{data_product_name}'"
-        logger.info(msg)
-        return format_response_json(HTTPStatus.OK, {"message": msg})
+        glue_client.delete_database(Name=data_product_name)
+    except glue_client.exceptions.EntityNotFoundException:
+        logger.info(f"Glue database '{data_product_name}' not found.")
+
+    # Delete fail files
+    s3_recursive_delete(get_fail_data_bucket(), [f"fail/{data_product_name}/"])
+    # Delete raw files
+    s3_recursive_delete(get_raw_data_bucket(), [f"raw/{data_product_name}/"])
+    # Delete curated files
+    s3_recursive_delete(get_curated_data_bucket(), [f"curated/{data_product_name}/"])
+    # Delete Metadata & Schema files
+    s3_recursive_delete(get_metadata_bucket(), [f"{data_product_name}/"])
```

Comment:
Can you handle any exceptions raised if there are errors during delete?
Perhaps a loop over a list of tuples of the method and the path 
```
error_msg = ""
for get_bucket_method, product_path in
[
    (get_fail_data_bucket(), [f"fail/{data_product_name}/"],
    (get_raw_data_bucket(), [f"raw/{data_product_name}/"],
    (get_curated_data_bucket(), [f"curated/{data_product_name}/"],
    (get_metadata_bucket(), [f"{data_product_name}/"],
]:
    try:
        s3_recursive_delete(get_bucket_method, product_path)
    except Exception as e:
        error_msg.append(e)
```

Reply:
No need as the function wont throw an exception unless the bucket it not found.https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/bucket/objects.html#delete

Reply:
Do we need to check the response from delete though to make sure nothing failed?

> For each key, Amazon S3 performs a delete action and returns the result of that delete, success, or failure, in the response.